File: car-sales-app/models/sale.js


class Sale {
    constructor(id, idClient, idCar, idEmployee, saleDate, finalPrice) {
        this.id = id;
        this.idClient = idClient;
        this.idCar = idCar;
        this.idEmployee = idEmployee;
        this.saleDate = new Date(saleDate);
        this.finalPrice = finalPrice;

        // Для отображения
        this.clientName = '';
        this.carName = '';
        this.employeeName = '';
    }

    setClientName(name) {
        this.clientName = name;
    }

    setCarName(name) {
        this.carName = name;
    }

    setEmployeeName(name) {
        this.employeeName = name;
    }

    static fromJson(json) {
        return new Sale(
            json.id_sale,
            json.id_client,
            json.id_car,
            json.id_employee,
            json.sale_date,
            json.final_price
        );
    }

    toJson() {
        const json = {
            id_sale: this.id,
            id_client: this.idClient,
            id_car: this.idCar,
            id_employee: this.idEmployee,
            sale_date: this.saleDate.toISOString().split('T')[0],
            final_price: this.finalPrice
        };
        console.log('📝 Обновляем продажу: ', json);
        return json;
    }
}

module.exports = Sale;

File: lib/models/sale.dart
class Sale {
  final int id;
  final int idClient;
  final int idCar;
  final int idEmployee;
  final DateTime saleDate;
  final double finalPrice;
  String clientName = '';
  String carName = '';
  String employeeName = '';
  final String? imagePath;

  Sale({
    required this.id,
    required this.idClient,
    required this.idCar,
    required this.idEmployee,
    required this.saleDate,
    required this.finalPrice,
    this.imagePath,
  });

  // Метод для создания объекта из JSON
  factory Sale.fromJson(Map<String, dynamic> json) {
    return Sale(
      id: json['id_sale'] ?? 0,
      idClient: json['id_client'] ?? 0,
      idCar: json['id_car'] ?? 0,
      idEmployee: json['id_employee'] ?? 0,
      saleDate: DateTime.tryParse(json['sale_date'] ?? '') ?? DateTime.now(),
      finalPrice: (json['final_price'] != null)
          ? double.tryParse(json['final_price'].toString()) ?? 0.0
          : 0.0,
      imagePath: json['image_path'] ?? json['imagePath'],
    );
  }

  // Метод для преобразования объекта в JSON
  Map<String, dynamic> toJson() {
    return {
      'id_sale': id,
      'id_client': idClient,
      'id_car': idCar,
      'id_employee': idEmployee,
      'sale_date': saleDate.toIso8601String().split('T').first,
      'final_price': finalPrice,
      'image_path': imagePath,
    };
  }

  void setClientName(String name) => clientName = name;
  void setCarName(String name) => carName = name;
  void setEmployeeName(String name) => employeeName = name;
}
